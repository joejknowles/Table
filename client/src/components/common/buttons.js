import React from 'react';
import { goToPath } from '../../helpers/routing';

export const ToPathButton = (props) => (
  <button
    onClick={ () => goToPath(props.path) }
    className={ props.className }
    >
    { props.res }
  </button>
);
