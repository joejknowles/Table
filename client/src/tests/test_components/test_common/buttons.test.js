import { DumbDispatchButton } from '../../../components/common/buttons';

import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';
import ReactTestUtils from 'react-addons-test-utils';
const shallowRenderer = ReactTestUtils.createRenderer();

it('DispatchButton renders as before', () => {
  const props = {
    res: 'test res', className: 'test class'
  };
  const component = shallowRenderer.render(
    <DumbDispatchButton { ...props } />
  );
  expect(component).toMatchSnapshot();
});

it('DispatchButton calls dispatch on click', () => {
  const fn = jest.fn();
  const action = { type: 'test' };
  const component = shallow(
    <DumbDispatchButton dispatch={ fn } />
  );
  expect(fn).not.toHaveBeenCalled();
  component.simulate('click');
  expect(fn).toHaveBeenCalled();
});
