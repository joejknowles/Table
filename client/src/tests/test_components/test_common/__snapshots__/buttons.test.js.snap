exports[`test DispatchButton renders as before 1`] = `
<button
  className="test class"
  onClick={[Function]}>
  test res
</button>
`;
